report/src/system_design/ble_implementation.tex

```diff
@@ -1,57 +1,27 @@
-The main page then displays the current status of whether the monitored devices are inside or outside the room, based on the calculated distances and the specified threshold (\todo{This last sentence may change in future}).
+When the application starts, the \texttt{Beacon Scanner} component will begin listening for Bluetooth devices and add those who match the iBeacon signature to a list of devices.
+These devices showed to the user in the view inside of a list.
+Each device in the list has a corresponding toggle, letting the user select which devices to use for data collection.
+After selecting at least one device, the user can click the "Start listening collection" button, which prompts the \texttt{Data Collector} class to listen for advertisement packets from the selected devices. 
+While the \texttt{Data Collector} is actively listening for packets, the \texttt{BeaconRssis} hashtable is continuously populated with the selected beacons' GUIDs and latest RSSI values.
 
-Once the data collection process is stopped by the user, the collected data is written to a JSON file on the device.
 
-% \subsection{Device Scanning and Selection}
-% The system employs the \texttt{BLEScannerService} to efficiently identify and manage available BLE devices within the proximity. This service initiates the device scanning process, incorporating specific configurations to optimize the scan, such as employing a low-latency scan mode and an infinite scan timeout. As a result, the service detects nearby devices with minimal delay and continuously updates the list of devices.
+At this point, the data collection process for a given point in the grid may begin.
+First, the user must select a label (inside or outside), denoting whether the current point is within or outside the room.
+Then, when the user presses the "Collect data" button on the view, the \texttt{Data Collector} class will begin collecting data from the selected beacons.
+The data collection process is implemented using standard deviation to account for fluctations.
+During data collection, the latest beacon RSSI measurements are added to a collection every 500ms until at least 40 measurements have been made, and the standard deviation is lower than some preset threshold.
+The collection interval was selected based on a heuristic -- the beacons advertise every 100ms, and therefore 500ms was selected to account for potential delays.
+Originally, the threshold was set to 20 measurements based on \citeauthor{improving_indoor_localization}~\cite{improving_indoor_localization}, which collected data for 10 seconds.
+Since we collect every 500ms, we could collect 20 measurements in ~10 seconds. However, during development, we found that the standard deviation varied too much with only 20 measurements, and therefore we doubled the minimum number of required measurements to 40.
```

Comment:
```suggestion
Since we collect every 500ms, we could collect 20 measurements in approximately 10 seconds. However, during development, we found that the standard deviation varied too much with only 20 measurements, and therefore we doubled the minimum number of required measurements to 40.
```